### ISSUE TITLE
linter: jsx-a11y/control-has-associated-label false positive when input is inside label

### ISSUE THREAD

**Issue by xt0rted** [oxc-project/oxc#22606]
### What version of Oxlint are you using?

1.66.0

### What command did you run?

oxlint

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

Seen using the default config on the Oxc Playground link from the rule page https://oxc.rs/docs/guide/usage/linter/rules/jsx_a11y/control-has-associated-label.html#references

### What happened?

An input inside of a label is flagged as not associated with a label. This contradicts the example code on the rule page that says this is the correct way to do it.

```html
<label>Name <input type="text" /></label>
```

Rule: https://oxc.rs/docs/guide/usage/linter/rules/jsx_a11y/control-has-associated-label.html
Error example: [https://playground.oxc.rs/?lintRules=jsx-a11y%2Fcontrol-has-associated-label](https://playground.oxc.rs/?lintRules=jsx-a11y%2Fcontrol-has-associated-label&options=%7B%22run%22%3A%7B%22lint%22%3Atrue%2C%22formatter%22%3Afalse%2C%22transform%22%3Afalse%2C%22isolatedDeclarations%22%3Afalse%2C%22whitespace%22%3Afalse%2C%22mangle%22%3Afalse%2C%22compress%22%3Afalse%2C%22scope%22%3Atrue%2C%22symbol%22%3Atrue%2C%22cfg%22%3Afalse%7D%2C%22parser%22%3A%7B%22extension%22%3A%22tsx%22%2C%22allowReturnOutsideFunction%22%3Atrue%2C%22preserveParens%22%3Atrue%2C%22allowV8Intrinsics%22%3Atrue%2C%22semanticErrors%22%3Atrue%7D%2C%22linter%22%3A%7B%22config%22%3A%22%7B%5C%22categories%5C%22%3A%7B%5C%22correctness%5C%22%3A%5C%22off%5C%22%7D%2C%5C%22rules%5C%22%3A%7B%5C%22jsx-a11y%2Fcontrol-has-associated-label%5C%22%3A%5C%22error%5C%22%7D%2C%5C%22plugins%5C%22%3A%5B%5C%22eslint%5C%22%2C%5C%22oxc%5C%22%2C%5C%22typescript%5C%22%2C%5C%22unicorn%5C%22%2C%5C%22jsx-a11y%5C%22%5D%7D%22%7D%2C%22formatter%22%3A%7B%22useTabs%22%3Afalse%2C%22tabWidth%22%3A2%2C%22endOfLine%22%3A%22lf%22%2C%22printWidth%22%3A80%2C%22singleQuote%22%3Afalse%2C%22jsxSingleQuote%22%3Afalse%2C%22quoteProps%22%3A%22as-needed%22%2C%22trailingComma%22%3A%22all%22%2C%22semi%22%3Atrue%2C%22arrowParens%22%3A%22always%22%2C%22bracketSpacing%22%3Atrue%2C%22bracketSameLine%22%3Afalse%2C%22objectWrap%22%3A%22preserve%22%2C%22singleAttributePerLine%22%3Afalse%7D%2C%22transformer%22%3A%7B%22target%22%3A%22es2015%22%2C%22useDefineForClassFields%22%3Atrue%2C%22experimentalDecorators%22%3Atrue%2C%22emitDecoratorMetadata%22%3Atrue%2C%22optimizeEnums%22%3Atrue%2C%22optimizeConstEnums%22%3Atrue%7D%2C%22isolatedDeclarations%22%3A%7B%22stripInternal%22%3Afalse%7D%2C%22codegen%22%3A%7B%22normal%22%3Atrue%2C%22jsdoc%22%3Atrue%2C%22annotation%22%3Atrue%2C%22legal%22%3Atrue%7D%2C%22compress%22%3A%7B%7D%2C%22mangle%22%3A%7B%22topLevel%22%3Atrue%2C%22keepNames%22%3Afalse%7D%2C%22controlFlow%22%3A%7B%22verbose%22%3Afalse%7D%2C%22inject%22%3A%7B%22inject%22%3A%7B%7D%7D%2C%22define%22%3A%7B%22define%22%3A%7B%7D%7D%7D&code=import+React%2C+%7B+useEffect%2C+useRef+%7D+from+%27react%27%0A%0Aconst+DummyComponent%3A+React.FC+%3D+%28%29+%3D%3E+%7B%0A++const+ref+%3D+useRef%3CHTMLInputElement%3E%28null%29%0A%0A++useEffect%28%28%29+%3D%3E+%7B%0A++++if+%28ref.current%29+ref.current.focus%28%29%0A++%7D%2C+%5B%5D%29%0A%0A++return+%28%0A++++%3Cdiv%3E%7BBoolean%28ref.current%29+%3F%3F+%28%0A++++++%3Clabel%3EName+%3Cinput+type%3D%22text%22+ref%3D%7Bref%7D+%2F%3E%3C%2Flabel%3E%0A++++%29%7D%0A++++%3C%2Fdiv%3E%0A++%29%0A%7D%0A%0Aexport+default+DummyComponent)

**Comment by mehm8128** [comment]
@camc314 @camchenry 
This is because [we removed default ignoreElements](https://github.com/oxc-project/oxc/pull/21985/changes/d485f46b60c09cfd35b3617d1c446f024a46a774#diff-1b5d52b32dd7fe237e04ab520597def6c494791ab35b8809f39929e286bc5e81L72-L92).
Can we restore them as default config values?
https://github.com/oxc-project/oxc/pull/21985#issuecomment-4363575954

**Comment by xiaokhkh** [comment]
Opened a PR to fix this: https://github.com/oxc-project/oxc/pull/22737
